app/src/main/java/uk/ac/ebi/atlas/search/metadata/CellTypeWheelDao.java

```diff
@@ -81,7 +81,7 @@ public CellTypeWheelDao(SolrCloudCollectionProxyFactory proxyFactory) {
         singleCellAnalyticsCollectionProxy = proxyFactory.create(SingleCellAnalyticsCollectionProxy.class);
     }
 
-    public ImmutableList<ImmutableList<String>> facetSearchCtwFields(String searchTerm, String species) {
+    public ImmutableList<ImmutableList<String>> facetSearchCtwFields(String searchTerm, String species, boolean isSpeciesSearch) {
```

Comment:
We had this case in one of the Uncle Bob videos and probably also in the Venkat Subramaniam training, as well.
The code would be much cleaner and more understandable if we create 2 well named methods from this method without the boolean parameter.
That would also help you clean the above conditions in line 118 and 123.
Thanks